libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt

```diff
@@ -149,40 +146,9 @@ internal class CameraComponent constructor(
     }
 
     private fun updateCameraLocation() {
-        coroutineScope.launch {
-            combine(
-                store.select { it.location?.enhancedLocation },
-                store.select { it.navigation }
-            ) { location, navigationState ->
-                location?.let {
-                    viewportDataSource.onLocationChanged(it)
-                    viewportDataSource.evaluate()
-                    if (!isCameraInitialized) {
-                        when (navigationState) {
-                            NavigationState.ActiveNavigation,
-                            NavigationState.Arrival -> {
-                                navigationCamera.requestNavigationCameraToFollowing(
-                                    stateTransitionOptions = NavigationCameraTransitionOptions
-                                        .Builder()
-                                        .maxDuration(0) // instant transition
-                                        .build()
-                                )
-                                store.dispatch(SetCameraMode(TargetCameraMode.Following))
-                            }
-                            else -> {
-                                navigationCamera.requestNavigationCameraToOverview(
-                                    stateTransitionOptions = NavigationCameraTransitionOptions
-                                        .Builder()
-                                        .maxDuration(0) // instant transition
-                                        .build()
-                                )
-                                store.dispatch(SetCameraMode(TargetCameraMode.Overview))
-                            }
-                        }
-                    }
-                    isCameraInitialized = true
```

Comment:
Apart from resetting camera state on view recreations this removed logic also made the first camera update instant. Was there a requirement for this? I think it looks better when the camera update is animated.

Reply:
Yeah this was done, otherwise the app always started at null island and then transitioned to the current location as can be seen in the video below.

https://user-images.githubusercontent.com/9770186/195206662-6b6caf01-6fe8-4665-b9f0-0a894adf3a0c.mp4

P.S: I like it personally tbh, but it seemed like this might be too much every time someone opens the app. cc @kmadsen

Reply:
Alright, I restored that logic, but updated it so that it doesn't reset the camera state to a default one for the current state. This also should fix the behavior you reported in https://github.com/mapbox/mapbox-navigation-android/pull/6472#issuecomment-1275329072.